### PR TITLE
FuzzyPrompt2

### PR DESCRIPTION
**Pull request by lbesnard**
- comicrack: more generic way to deduce archive format, not based on extension
- bdgest:add interactive fuzzy prompt when album not matched automatically

### CODE REVIEW

File: bdnex/lib/bdgest.py [hunk 1]
```diff
@@ -397,12 +419,27 @@ def parse_album_metadata_mobile(self, album_name):
         if isinstance(album_meta_dict['Planches'], str):
             album_meta_dict['Planches'] = int(album_meta_dict['Planches'])
 
-        if hasattr(album_meta_dict, 'Tome'):
+        if 'Tome' in album_meta_dict.keys():
```
Comment (lbesnard) [inline bdnex/lib/bdgest.py:444]: @jpmenil , sorry for the messy PR doing too many things in one. This is similar to https://github.com/lbesnard/bdnex/pull/13

Comment (jpmenil) [reply bdnex/lib/bdgest.py:444]: :)

**Comment by jpmenil** [comment]
So, now you extract the content and re-create the archive, whatever is the extension, right?
which may be a bit slower than before.
The only way i know, to not deal with file extensions, is to rely on the magic number of the file header:
```
header_type = i.read(4).hex()
if header_type == '52617221':
    print('RAR', i.name)
elif header_type == '504b0304':
    print('PKz', i.name)
elif header_type == '25504446':
     print('PDF', i.name)
else:
    print('Unknown', i.name)
```

i don't have strong opinions on this :)

**Comment by jpmenil** [comment]
Just to let you know, that with patool and cbr files, i often hit this issue:
https://github.com/wummel/patool/issues/117
I can remediate it by forcing the rar program to 7z:
```
patoolib.extract_archive(self.input_filename, interactive=False, outdir=extracted_dir, program='7z')
```
but it only work with this fix:
https://github.com/wummel/patool/pull/101

**Comment by lbesnard** [comment]
> Just to let you know, that with patool and cbr files, i often hit this issue:

yeah this is annoying. I remediate it by just using rarfile for rar files.

If you test this, let me know how you go with archives containing Latin-1 characters. I found some issues with some linux generic cbz/cbr viewers, but not with komga. My env is on UTF-8

**Comment by jpmenil** [comment]
will test.
And you handle the fact that a previous comicinfo.xml exist, nice.
I used to spawn a subprocess in order to delete it. (i learned that you can append multiple time the same file in a zip file ...)

**Review by jpmenil — APPROVED** [review]

**Pull request merged**